### PR TITLE
Optimize concrete runtime contract code

### PR DESCRIPTION
**Pull request by arcz**
## Description
This PR optimizes runtime contract code that is known to be concrete. It resulted in a significant performance bump when executing ethereum-tests suite on my machine down to 76s from 86s (~13%).

## Checklist

- [x] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [ ] updated the changelog

### CODE REVIEW

File: src/EVM.hs [hunk 1]
```diff
@@ -305,10 +305,15 @@ data SubState = SubState
   hopefully we do not have to deal with dynamic immutable before we get a real data section...
 -}
 data ContractCode
-  = InitCode ByteString (Expr Buf)     -- ^ "Constructor" code, during contract creation
-  | RuntimeCode (V.Vector (Expr Byte)) -- ^ "Instance" code, after contract creation
+  = InitCode ByteString (Expr Buf) -- ^ "Constructor" code, during contract creation
+  | RuntimeCode RuntimeCode -- ^ "Instance" code, after contract creation
   deriving (Show)
 
+data RuntimeCode
```
Comment (d-xo) [inline src/EVM.hs:312]:
```suggestion
-- | We have two variants here to optimize the fully concrete case.
-- ConcreteRuntimeCode just wraps a ByteString
-- SymbolicRuntimeCode is a fixed length vector of potentially symbolic bytes, which lets us handle symbolic pushdata (e.g. from immutable variables in solidity).
data RuntimeCode
```

File: src/EVM.hs [hunk 2]
```diff
@@ -1075,7 +1079,9 @@ exec1 = do
           case stk of
             (x:xs) ->
               burn g_mid $ forceConcrete x "JUMP: symbolic jumpdest" $ \x' ->
-                checkJump x' xs
```
Comment (d-xo) [inline src/EVM.hs:1078]: why remove the `checkJump` here?

**Review by d-xo — COMMENTED** [review]
Looks great generally, just a couple of nits & questions.

File: src/EVM.hs (hunk 2 above)
Comment (arcz) [reply src/EVM.hs:1078]: it's still there but guarded under the case

Comment (d-xo) [reply src/EVM.hs:1078]: oh of course... :woman_facepalming:

**Review by d-xo — APPROVED** [review]

**Pull request merged**